internal/mode/static/state/graph/nginxproxy.go

```diff
@@ -7,18 +7,59 @@ import (
 	ngfAPI "github.com/nginxinc/nginx-gateway-fabric/apis/v1alpha1"
 )
 
+type NginxProxy struct {
```

Comment:
I'm probably missing something, but why do we need these NginxProxy and Tracing struct definitions? Can't we use the ones from ngfAPI here?

Reply:
I wanted to simplify where we had to do the nested nil-checking, but also now with the validation code, I think it makes sense to have this is a separate type. What do you think?

Reply:
I see two pros for removing it:
- for consistency with other types - we don't do the same thing for other types
- although we will need to do nil checking, we're not gonna do it twice - we can convert NginxProxy into dataplane.Tracing when building dataplane.Configuration

Reply:
My concern is around validation and building the status. I'd need to do something different here than what we do for the other resources, because I'm doing the validation here and need to propagate the conditions to the status updater.